cocos/ui/UIWidget.cpp

```diff
@@ -558,6 +558,10 @@ void Widget::setTouchEnabled(bool enable)
     if (_touchEnabled)
     {
         _touchListener = EventListenerTouchOneByOne::create();
+        if (!_touchListener) {
```

Comment:
In modern operating system, we will never come across "run out of memory exception"? 
When the memory is not enough, os will use external storage and paging occurs.

Am I right? If not, please correct me.

And in mobile platform, every apps are running inside of a sandbox, if the apps consumes many memory, the os will kill it. So we don't need to use assert here.
If we do want to check "run out of memory issue", maybe this kind of checking code will scatter everywhere in the code base.